x-pack/elastic-agent/pkg/agent/program/testdata/single_config-filebeat.yml

```diff
@@ -21,6 +20,26 @@ filebeat:
             type: logs
             dataset: generic
             namespace: default
+  - type: log
+    paths:
+      - /var/log/hello3.log
+      - /var/log/hello4.log
+    index: testtype-generic-default
+    vars:
+      var: value
+    processors:
+      - add_fields:
+          target: "dataset"
+          fields:
+            type: testtype
+            name: generic
+            namespace: default
+      - add_fields:
```

Comment:
Not related to this PR but can you follow up to make sure we clean up the stream fields?